bundler/lib/bundler/source/git/git_proxy.rb

```diff
@@ -250,6 +251,10 @@ def pinned_to_full_sha?
           ref =~ /\A\h{40}\z/
         end
 
+        def full_sha_revision?
+          @revision.match?(/\A\h{40}\z/)
+        end
```

Comment:
This logic seems mainly duplicated with `pinned_to_full_sha?`. I would refactor this logic to a private method that takes a `ref` and does this regexp match, and then rename this to `locked_to_full_sha?`, and make both `pinned_to_full_sha?` and `locked_to_full_sha?` use the private method.

Reply:
Thank you for the review.

I fixed f0d8255ebd89f2dd47a53fb400c5ed6a61f72c28

Reply:
Thank you!